miscellaneous/metaDataRemover/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-tools-go/miscellaneous/metaDataRemover/config"
+	"github.com/ElrondNetwork/elrond-tools-go/trieTools/trieToolsCommon"
+	"github.com/pelletier/go-toml"
+	"github.com/urfave/cli"
+)
+
+const (
+	logFilePrefix   = "meta-data-remover"
+	tomlFile        = "./config.toml"
+	outputFilePerms = 0644
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "Tokens exporter CLI app"
+	app.Usage = "This is the entry point for the tool that deletes tokens meta-data"
+	app.Flags = getFlags()
+	app.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+
+	app.Action = func(c *cli.Context) error {
+		return startProcess(c)
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+		return
+	}
+}
+
+func startProcess(c *cli.Context) error {
+	flagsConfig := getFlagsConfig(c)
+
+	_, errLogger := trieToolsCommon.AttachFileLogger(log, logFilePrefix, flagsConfig.ContextFlagsConfig)
+	if errLogger != nil {
+		return errLogger
+	}
+
+	log.Info("sanity checks...")
+
+	err := logger.SetLogLevel(flagsConfig.LogLevel)
+	if err != nil {
+		return err
+	}
+
+	log.Info("starting processing", "pid", os.Getpid())
+
+	shardTokensMap, err := readTokensInput(flagsConfig.Tokens)
+	if err != nil {
+		return err
+	}
+
+	cfg, err := loadConfig()
+	if err != nil {
+		return err
+	}
+
+	shardTxsDataMap, err := createShardTxsDataMap(shardTokensMap, cfg.TokensToDeletePerTransaction)
+	if err != nil {
+		return err
+	}
+
+	shardPemsDataMap, err := readPemsData(flagsConfig.Pems)
+	if err != nil {
+		return err
+	}
+
+	return createShardTxs(flagsConfig.Outfile, cfg, shardPemsDataMap, shardTxsDataMap)
+}
+
+func readTokensInput(tokensFile string) (map[uint32]map[string]struct{}, error) {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	fullPath := filepath.Join(workingDir, tokensFile)
+	jsonFile, err := os.Open(fullPath)
+	if err != nil {
+		return nil, err
+	}
+
+	bytesFromJson, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+
+	shardTokensMap := make(map[uint32]map[string]struct{})
+	err = json.Unmarshal(bytesFromJson, &shardTokensMap)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info("read from input", "file", tokensFile, "num of shards", len(shardTokensMap), getNumTokens(shardTokensMap))
+	return shardTokensMap, nil
+}
+
+func loadConfig() (*config.Config, error) {
+	tomlBytes, err := ioutil.ReadFile(tomlFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var cfg config.Config
+	err = toml.Unmarshal(tomlBytes, &cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cfg, nil
+}
+
+func readPemsData(pemsFile string) (map[uint32]*pkAddress, error) {
```

Comment:
we need to redesign this. From the experience, we won't have .pems for signing the transactions, it will likely be a ledger behind the mnemonics. Suggestion to somehow format the transaction, prepare the gas price/gas limit, set the value to 0, set the correct address and then, someone else will just sign the transaction on another computer. Maybe @andreibancioiu will guide us how this was done in the erdpy

Reply:
Unfortunately, this would not be feasible, since in case of removing 100 nonces/tx, with a volume of 7_000_000 nonces, we would need to sign 70_000 txs with the ledger

Reply:
right, will remain as it is. Will figure out how to properly manage this.